src/hotspot/share/prims/resolvedMethodTable.cpp

```diff
@@ -53,7 +53,8 @@ static const size_t GROW_HINT = 32;
 static const size_t ResolvedMethodTableSizeLog = 10;
 
 unsigned int method_hash(const Method* method) {
-  unsigned int hash = method->klass_name()->identity_hash();
+  unsigned int hash = method->method_holder()->class_loader_data()->identity_hash();
```

Comment:
I don't see "#include classfile/classLoaderData.hpp" in your patch. Did you verify it builds successfully? I saw a compilation failure without the include:

```
src/hotspot/share/prims/resolvedMethodTable.cpp:56:67: error: member access into incomplete type 'ClassLoaderData'
  unsigned int hash = method->method_holder()->class_loader_data()->identity_hash();
                                                                  ^
src/hotspot/share/memory/allocation.hpp:260:7: note: forward declaration of 'ClassLoaderData'
class ClassLoaderData;
      ^
1 error generated.
```

Reply:
Sorry, I only verify it with normal build parameters, and I forgot that "precompiled.hpp" might be disabled.